layouts/fastly/helix.vcl

```diff
@@ -387,13 +387,18 @@ sub vcl_recv {
           set req.backend = F_GitHub;
         }
       }
-      set req.url = "/" + var.owner + "/" + var.repo + "/" + var.ref + "/" + var.dir + "/" + req.url.basename + "?" + req.url.qs;
+
+      set var.path = var.dir + "/" + req.url.basename;
+      set req.url = "/" + var.owner + "/" + var.repo + "/" + var.ref + var.path + "?" + req.url.qs;
     } else {
+      set var.path = var.dir + "/" + var.name + ".md";
       # Invoke OpenWhisk
-      set req.url = "/api/v1/web" + var.action + "?owner=" + var.owner + "&repo=" + var.repo + "&ref=" + var.ref + "&path=" + var.dir + "/" + var.name + ".md" + "&selector=" + var.selector + "&extension=" + req.url.ext + "&branch=" + var.branch + "&strain=" + var.strain + "&GITHUB_KEY=" + table.lookup(secrets, "GITHUB_TOKEN");
+      set req.url = "/api/v1/web" + var.action + "?owner=" + var.owner + "&repo=" + var.repo + "&ref=" + var.ref + "&path=" + var.path + "&selector=" + var.selector + "&extension=" + req.url.ext + "&branch=" + var.branch + "&strain=" + var.strain + "&GITHUB_KEY=" + table.lookup(secrets, "GITHUB_TOKEN");
     }
   }
 
+  set req.url = regsuball(req.url, "/+", "/");
```

Comment:
This can become a problem when we introduce URL parameters, where `//` is a completely acceptable value. Probably better to just sanitize `var.path`.